### PR TITLE
Add ambassador as an exposer

### PR DESCRIPTION
**Pull request by heroic**
This adds [Ambassador](http://getambassador.io) as a supported exposer.

### CODE REVIEW

**Comment by heroic** [comment]
How do I see what test failed? And the logs?

**Comment by pmuir** [comment]
```[Pipeline] node
Running on Jenkins in /tmp/jenkinsTests.tmp/jenkins2339810596073617348test/workspace/job
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Declarative: Checkout SCM)
[Pipeline] checkout
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (CI Build and Test)
[Pipeline] dir
Running in /home/jenkins/go/src/github.com/jenkins-x/exposecontroller
[Pipeline] {
[Pipeline] checkout
[Pipeline] sh
+ make test
mkdir -p /home/jenkins/go/src/github.com/jenkins-x/exposecontroller/_gopath/src/github.com/jenkins-x
ln -s -f /home/jenkins/go/src/github.com/jenkins-x/exposecontroller /home/jenkins/go/src/github.com/jenkins-x/exposecontroller/_gopath/src/github.com/jenkins-x
CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -ldflags " -X github.com/jenkins-x/exposecontroller/version.Version='2.3.35' -X github.com/jenkins-x/exposecontroller/version.Revision='f34123c7' -X github.com/jenkins-x/exposecontroller/version.Branch='master' -X github.com/jenkins-x/exposecontroller/version.BuildUser='@82607ba2-3248-11e9-820f-0a580a305a04-pod-444913' -X github.com/jenkins-x/exposecontroller/version.BuildDate='20190217-00:11:56' -X github.com/jenkins-x/exposecontroller/version.GoVersion='1.11.5' -s -w -extldflags '-static'" -o ./out/exposecontroller-linux-amd64 github.com/jenkins-x/exposecontroller
_gopath/src/github.com/jenkins-x/exposecontroller/exposestrategy/ambassador.go:12:2: cannot find package "gopkg.in/yaml.v2" in any of:
	/home/jenkins/go/src/github.com/jenkins-x/exposecontroller/_gopath/src/github.com/jenkins-x/exposecontroller/vendor/gopkg.in/yaml.v2 (vendor tree)
	/usr/local/go/src/gopkg.in/yaml.v2 (from $GOROOT)
	/home/jenkins/go/src/github.com/jenkins-x/exposecontroller/_gopath/src/gopkg.in/yaml.v2 (from $GOPATH)
Makefile:62: recipe for target 'out/exposecontroller-linux-amd64' failed
make: *** [out/exposecontroller-linux-amd64] Error 1
[Pipeline] }
[Pipeline] // dir
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build and Release)
Stage "Build and Release" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE```

**Comment by jstrachan** [comment]
we've unfortunately not moved exposecontroller over to go 11 modules yet btw.

For now we've been importing `gopkg.in/v2/yaml` - lemme try change the import and see if that works

**Comment by jstrachan** [comment]
just fixing up a couple more compile errors hopefully...

File: exposestrategy/ambassador.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	} else {
+		return errors.Wrapf(err, "failed to patch the service %s/%s", svc.Namespace, appName)
+	}
+}
+
+func (s *AmbassadorStrategy) Remove(svc *api.Service) error {
+	delete(svc.Annotations, "getambassador.io/config")
+
+	_, err := s.client.Services(svc.Namespace).Update(svc)
+	if err != nil {
```
Comment (polothy) [inline exposestrategy/ambassador.go]: Should this be `==` instead of `!=`?

Comment (jstrachan) [reply exposestrategy/ambassador.go]: great catch - lemme fix that!

Comment (jstrachan) [reply exposestrategy/ambassador.go]: PR to fix that here: https://github.com/jenkins-x/exposecontroller/pull/170

**Pull request merged**